package.json

```diff
@@ -10,6 +10,11 @@
   "repository": {
     "private": true
   },
+  "overrides": {
+    "es-check@^5.2.4": {
+      "cli-table3": "^0.6.1"
+    }
+  },
```

Comment:
If I remove this and `npm i`, `cli-table3` is downgraded but there still are no `npm audit` warnings.

* [ ] Please check if we really need this

Reply:
I think this is a bug in `npm`. The `overrides` flag seems to be quite new and tools like `npm audit fix` do not (yet) make use of it. If you remove the `overrides` flag as described and reinstall `es-check` (`npm rm es-check && npm i es-check@^5.2.4`) the `npm audit` warning will reappear.

Reply:
How did we create the package-lock? If I undo the change to the lockfile and run `npm i` (npm v.8.1.0) the vulnerabilities don't go away.

Reply:
`npm rm es-check && npm i es-check@^5.2.4`

Npm still seams to only in same cases make use of the `overrides` feature. E.g. `npm audit fix` does not make use of `overrides` to fix the issue.

Reply:
OK I now have it working as described with npm 8.5.2

I see that we expressed the intention in a very specific way: please patch the unsecure `cli-table3` for a specific package and version.

How about a different approach, given we know that a certain `cli-table3` should not be used at all:

```json
  "overrides": {
    "cli-table3": ">=0.6.1"
  },
```
This would make sure every `cli-table3` dependency uses a safe version, leaving dependencies to good versions intact.

* [ ] As this is a new feature I don't know yet which usage is best practice. WDYT?

Reply:
I like the idea. But I think npm is not smart enough (yet?). E.g. if a differente package get's installed, that has a dependency on `cli-table3` version `^1.0.0` (which does not exist) npm would now be able to install version `0.6.1` of that package, because that is in the range of `>=0.6.1`. Also if `cli-table3` version `1.0.0` would get released with a breaking API change, all packages, that rely on the 0.x api could break.

So given that, I would prefer to limit `overrides` as much as possible to avoid messing with other packages as well.

Reply:
> npm would now be able to install version 0.6.1 of that package

That would be bad. When I tested it, npm preferred the default over override. Is the expected behavior documented somewhere?

Reply:
I found the following documentation:
* Official documentation: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides
* First implementation: https://github.com/npm/arborist/pull/354
* Spec?: https://github.com/npm/rfcs/blob/main/accepted/0036-overrides.md

Reply:
Nice. As the spec (for me) doesn't answer how and if ranges of dependencies and overrides interact, how about making the intention explicit:
```json
  "overrides": {
    "cli-table3@<0.6.1": "^0.6.1"
  },
```

This would match
> it is arguably better in most cases to apply the override too broadly rather than too narrowly.

and clearly express the idea
> Using this feature should be considered a hack in most cases, something that is done temporarily while waiting for a bug to be fixed